lib/facter/custom_facts/core/execution/base.rb

```diff
@@ -114,7 +114,7 @@ def extract_options(options)
           on_fail = options.fetch(:on_fail, :raise)
           expand = options.fetch(:expand, true)
           logger = options[:logger]
-          time_limit = options[:limit].to_i
+          time_limit = options[:time_limit].to_i
```

Comment:
Other things we can do:

1. See if `:limit` is set and if so use if as `:time_limit` and emit a warning;
2. Emit a warning for each unexpected key in the hash.

Reply:
1 makes sense, with the mention that it would be better to use `timeout` (as per my comment below).

2 also makes sense, we can have a list of deprecated options (which would be only `limit` as of now) and a list of accepted options: `timeout`, `expand`, `on_fail`, `logger` (not 100% sure what this does). If the user sets a deprecated option we should accept it and warn, and if the user sets something else we should warn.

It would be good to have a ticket created for this change. Let me know if you can work on this @smortex